oracle/types/market.go

```diff
@@ -1,160 +1,94 @@
 package types
 
 import (
-	"encoding/json"
 	"fmt"
-	"os"
 
-	mmtypes "github.com/skip-mev/slinky/x/marketmap/types"
+	"github.com/skip-mev/slinky/pkg/json"
+	mmtypes "github.com/skip-mev/slinky/x/mm2/types"
 )
 
-type (
-	// TickerToProviderConfig is a type alias for the map of tickers to their respective
-	// provider configurations.
-	TickerToProviderConfig = map[mmtypes.Ticker]mmtypes.ProviderConfig
-
-	// ProviderMarketMap provides a simple adapter of what the market map module expects
-	// from a provider and how the provider configures its markets. It provides a bi-directional
-	// mapping between on-chain and off-chain markets. Every ProviderMarketMap is expected to be
-	// constructed from a given market map module configuration.
-	ProviderMarketMap struct {
-		// Name is the name of the provider.
-		Name string
-
-		// TickerConfigs is a map of tickers to their respective off-chain markets as
-		// configured by the market map module.
-		TickerConfigs TickerToProviderConfig
-
-		// OffChainMap is a map of tickers to their respective on-chain markets.
-		OffChainMap map[string]mmtypes.Ticker
-	}
-)
-
-// ProviderMarketMapFromMarketMap returns a provider market map from a market map provided by the
-// market map module.
-func ProviderMarketMapFromMarketMap(name string, marketMap mmtypes.MarketMap) (ProviderMarketMap, error) {
+// ProviderTickersFromMarketMap returns the set of provider tickers a given provider should
+// be providing data for based on the market map.
+func ProviderTickersFromMarketMap(
+	name string,
+	marketMap mmtypes.MarketMap,
+) ([]ProviderTicker, error) {
 	if err := marketMap.ValidateBasic(); err != nil {
-		return ProviderMarketMap{}, fmt.Errorf("invalid market map: %w", err)
+		return nil, fmt.Errorf("invalid market map: %w", err)
 	}
 
-	// Iterate over the providers and their respective tickers.
-	tickers := make(TickerToProviderConfig)
-	for tickerStr, config := range marketMap.Providers {
-		ticker, ok := marketMap.Tickers[tickerStr]
-		if !ok {
-			return ProviderMarketMap{}, fmt.Errorf("ticker %s not found in market map", tickerStr)
-		}
-
-		for _, provider := range config.Providers {
-			if provider.Name != name {
+	var (
+		// Track all of the tickers that the provider will be providing data for.
+		providerTickers = make([]ProviderTicker, 0)
+		// Maintain a set of off-chain tickers that have been seen to avoid duplicates.
+		seenOffChainTickers = make(map[string]struct{})
+	)
+
+	// Iterate through every single market and its provider configurations to find the
+	// provider configurations that match the provider name.
+	for _, market := range marketMap.Markets {
```

Comment:
we can probably optimize this a bit in the future (right now we are doing O(n2) search and can probably make it O(n) with some caching or awareness between providers when constructing.)

not crucial now